x.ps1

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env pwsh
+
+# See x.sh for why these scripts exist.
+
+$xpy = $(Split-Path $MyInvocation.MyCommand.Path -Parent) + "/x.py"
```

Comment:
For powershell v3+ you can use `$PSScriptRoot`, I don't know what level of compatibility you're going for though

Reply:
Done, thanks! No particular level - if it works on ~the last year of Windows 10, that's good enough. Don't expect anyone to be building rust from source on Windows 7.

Reply:
Preinstalled Windows PowerShell is

| OS | PowerShell |
| 7 SP1 | 2.0 |
| Server 2008 R2 SP1 | 2.0 |
| 8 | 3.0 |
| Server 2012 | 3.0 |
| 8.1 | 4.0 |
| Server 2012 R2 | 4.0 |
| 10 | 5.0[^1] |
| Server 2016 | 5.0[^1] |
| 11 | 5.1 |

[^1]: Automatically updated to 5.1 by Windows Update

So the important thing is just working both in Windows PowerShell (reasonable to assume 5.0) and [PowerShell Core](https://github.com/PowerShell/PowerShell) (currently 7.2.5; also covers 6.* with previous releases). These are two separate installations on a Windows machine; Windows Powershell is there by default, and PowerShell Core is the cross-platform version of PowerShell that's open source, getting regular updates, and needs to be user installed. If we want to be extra careful, we can check against `$host.version.major` to print a warning if `< 5`, but that should be a very rare edge case at this point since Win7 is EOL without enterprise ESU.

Launching Windows PowerShell without `-NoLogo` prompts to [migrate to PowerShell Core](https://docs.microsoft.com/en-us/powershell/scripting/whats-new/migrating-from-windows-powershell-51-to-powershell-7?view=powershell-7.2).

`pwsh` is PowerShell Core, `powershell` is Windows PowerShell.